webwhatsapi/js/wapi.js

```diff
@@ -80,6 +80,23 @@ window.WAPI._serializeMessageObj = (obj) => {
     });
 };
 
+window.WAPI.getConnectionInfo = function (done) {
+    if (done !== undefined) {
+        done(document.querySelector("#app")._reactRootContainer.current.child.child.child.child.child.memoizedProps.children[5].props.conn);
+    } else {
+        return document.querySelector("#app")._reactRootContainer.current.child.child.child.child.child.memoizedProps.children[5].props.conn;
+    }
+}
+
+window.WAPI.getChatsModel = function (done) {
+    if (done !== undefined) {
+        done(document.querySelector("#app")._reactRootContainer.current.child.child.child.child.child.child.sibling.sibling.sibling.sibling.sibling.child.child.child.child.child.sibling.sibling.sibling.sibling.sibling.child.child.child.child.memoizedState.chats);
+    } else {
+        return document.querySelector("#app")._reactRootContainer.current.child.child.child.child.child.child.sibling.sibling.sibling.sibling.sibling.child.child.child.child.child.sibling.sibling.sibling.sibling.sibling.child.child.child.child.memoizedState.chats;
```

Comment:
Please change this to: 
return document.querySelector("#app")._reactRootContainer.current.child.child.child.child.child.child.sibling.sibling.sibling.sibling.sibling.child.child.child.child.child.sibling.sibling.sibling.sibling.sibling.child.child.child.child.memoizedState.chats[0].collection;

to obtain all chats, not only cached ones. Notice the "[0].collection" at the end.